mm2src/mm2_main/src/lp_swap/swap_v2_rpcs.rs

```diff
@@ -501,3 +503,170 @@ pub(crate) async fn active_swaps_rpc(
         statuses,
     })
 }
+
+#[derive(Deserialize)]
+pub(crate) struct StopSwapRequest {
+    uuid: Uuid,
+}
+
+#[derive(Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub(crate) enum StopSwapErr {
+    Internal(String),
+    NotRunning,
+}
+
+impl HttpStatusCode for StopSwapErr {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            StopSwapErr::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            StopSwapErr::NotRunning => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+#[derive(Serialize)]
+pub(crate) struct StopSwapResponse {
+    result: String,
+}
+
+pub(crate) async fn stop_swap_rpc(ctx: MmArc, req: StopSwapRequest) -> MmResult<StopSwapResponse, StopSwapErr> {
+    let swap_ctx = SwapsContext::from_ctx(&ctx).map_err(StopSwapErr::Internal)?;
+    let mut running_swaps = swap_ctx.running_swaps.lock().unwrap();
+    let Some(position) = running_swaps.iter().position(|(swap, _)| swap.upgrade().map_or(true, |swap| swap.uuid() == &req.uuid)) else {
+        return MmError::err(StopSwapErr::NotRunning);
+    };
+    let (_swap, _abort_handle) = running_swaps.swap_remove(position);
```

Comment:
When a swap is stopped like this, its future basically aborted.
Could we get inconsistent swap state files because of this (like we saw before: a spend tx was sent but the event was not written into the swap state file.

I think it would be more accurate to introduce a cancelling cmd.

Reply:
> Could we get inconsistent swap state files because of this?

Yes we would.

Cancelling from within would really be messy, as you need to involve the cancelling code inside the states and not just in the runner loop (e.g. if you try to cancel a swap that is now in recovery loop, u can't code that in the outer swap loop, but need to code this into the recovery loop itself otherwise the `stop` will stop after indetermined amount of time, same for any operation in every state that might take a lot of time).
I think we generally should write code in a way that we are sure things won't go ugly if we terminate at any specific line (even hardware failure) (e.g. store the data before we broadcast it etc...)

Reply:
> things won't go ugly if we terminate at any specific line (even hardware failure) (e.g. store the data before we broadcast it etc...)

This is not easy and generally provided by a 2-phase commit resource manager (ensures a message is sent and db is updated as an atomic transaction)

Reply:
As an example, in c++ boost they provide thread::interruption_point() which may be used to cancel a long running thread (for e.g. placing in a loop which does not yield control).
I thought maybe we could create something similar which allow to abort the swap in points of consistency, when tx is sent and events are stored.

Reply:
Then we need ensure that we are able to kickstart a swap after it was stopped with this rpc.
In #2283 we fixed a situation when payment spent tx status and events state become inconsistent.
But I guess there may be other cases too:
for e.g. what if taker sends taker fee and then is suddenly stopped, before updating the events in the swap file. If we restart this swap I guess it will try to re-send taker fee.
However I guess, such cases, occurring due to the stop rpc, would be very rare and more likely would happen because of app or config errors.
But as the rpc theoretically may lead to that, I suggest at least we log stop rpc call, for easier investigation.

Reply:
btw, the stop rpc was created for dev/qa usage. don't think we will include it in user facing apps.

Reply:
[997ad63](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2280/commits/997ad636f2c2accf93c490d5229cd3a32e85671a)